cmd/registry/auth/basic/basic.go

```diff
@@ -43,22 +52,52 @@ func NewBasicCmd(ctx context.Context, opt *options.Common) *cobra.Command {
 		Use:                   "basic [hostname]",
 		DisableFlagsInUseLine: true,
 		Short:                 "Login to an OCI registry",
-		Long:                  "Login to an OCI registry to push and pull artifacts",
-		Args:                  cobra.ExactArgs(1),
+		Long: `Login to an OCI registry
+
+Example - Log in with username and password from command line flags:
+	falcoctl registry auth basic -u username -p password localhost:5000
+
+Example - Login with username and password from env variables:
+	FALCOCTL_REGISTRY_AUTH_BASIC_USERNAME=username FALCOCTL_REGISTRY_AUTH_BASIC_PASSWORD=password falcoctl registry auth basic localhost:5000
+
+Example - Login with username and password from stdin:
+	falcoctl registry auth basic -u username --password-stdin localhost:5000
+
+Example - Login with username and password in an interactive prompt:
+	falcoctl registry auth basic localhost:5000
+`,
+		Args: cobra.ExactArgs(1),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			o.username = viper.GetString("registry.auth.basic.username")
+			o.password = viper.GetString("registry.auth.basic.password")
+			o.passwordFromStdin = viper.GetBool("registry.auth.basic.password_stdin")
+
```

Comment:
question, what if we also support automatically fallbacking at interactive prompt for password if only `-u` is passed (ie: if password is empty but username is not)?

Reply:
And we deprecated `--password-stdin`

Reply:
> question, what if we also support automatically fallbacking at interactive prompt for password if only `-u` is passed (ie: if password is empty but username is not)?

We already do that. If the password is not set using the -p, --password, or the env variable it falls back at the interactive prompt.

Reply:
> And we deprecate `--password-stdin`

The `--password-stdin` is used when piping the password from another command. For example:
```bash
echo $PASSWORD | ./falcoctl registry auth basic -u username localhost:5000 --password-stdin

```

Reply:
Ah makes sense, thanks!